### PR TITLE
Add net7 to testing matrix

### PR DESCRIPTION
**Pull request by halspang**
# Description

Add net7 into the testing framework to ensure compatibility.

Signed-off-by: halspang <halspang@microsoft.com>

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: NA

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: test/Dapr.Actors.AspNetCore.IntegrationTest/ActivationTests.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public class ActivationTests
         public async Task CanActivateActorWithDependencyInjection()
         {
             using var factory = new AppWebApplicationFactory();
-            var httpClient = factory.CreateClient();
+            var httpClient = factory.CreateClient(new Microsoft.AspNetCore.Mvc.Testing.WebApplicationFactoryClientOptions { HandleCookies = false });
```
Comment (rynowak) [inline test/Dapr.Actors.AspNetCore.IntegrationTest/ActivationTests.cs:36]: Does this work on downlevel .NET versions? eg: is this API present pre-7.0?

File: test/Dapr.Actors.AspNetCore.IntegrationTest/ActivationTests.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ private async Task<string> IncrementCounterAsync(HttpClient httpClient, string a
             var actorTypeName = nameof(DependencyInjectionActor);
             var methodName = nameof(DependencyInjectionActor.IncrementAsync);
 
-            var request = new HttpRequestMessage(HttpMethod.Put, $"http://localhost/actors/{actorTypeName}/{actorId}/method/{methodName}");
+            var request = new HttpRequestMessage(HttpMethod.Put, $"http://localhost:65326/actors/{actorTypeName}/{actorId}/method/{methodName}");
```
Comment (rynowak) [inline test/Dapr.Actors.AspNetCore.IntegrationTest/ActivationTests.cs:53]: What's the reason for this change?

**Comment by rynowak** [comment]
You should be able to also remove .NET 5 from your testing matrix as it was EOL in May 2022. This is not a breaking change since it doesn't break anything for a .NET 5 user. We'll just stop testing it.

File: test/Dapr.Actors.AspNetCore.IntegrationTest/ActivationTests.cs (hunk 1 above)
Comment (halspang) [reply test/Dapr.Actors.AspNetCore.IntegrationTest/ActivationTests.cs:36]: Yes, it is present for 3.1-7, though only 7 appears to handle cookies incorrectly. Given this comes from the integration suite, I believe it doesn't represent an issue with our standard code and I did manually test the actor example with net7 to ensure this isn't an issue.

File: test/Dapr.Actors.AspNetCore.IntegrationTest/ActivationTests.cs (hunk 2 above)
Comment (halspang) [reply test/Dapr.Actors.AspNetCore.IntegrationTest/ActivationTests.cs:53]: I added that when I was debugging the cookie issue. I'll remove it.

**Review by rynowak — APPROVED** [review]
